### PR TITLE
Profile refactor

### PR DESCRIPTION
**Pull request by tdunbar123**
Sorry for the long pull request! Will try to not allow this to get backed up in the future. To summarize, Andrew P added tests, and solution to making sure images weren't larger than 2Mb. Ashley wrote a test for adding User IDs and Diego wrote tests for adding followers and followees to domain profiles.

### CODE REVIEW

File: Fakebook.Profile/Fakebook.Profile.RestApi/Controllers/ProfilePictureController.cs [hunk 1]
```diff
@@ -57,6 +61,12 @@ public async Task<ActionResult> Post()
                 return BadRequest();
             }
 
+            if (file.Length > _maxFileSize)
+            {
+                _logger.LogError("File give to image was too large");
+                return BadRequest();
+            }
```
Comment (escalonn) [inline Fakebook.Profile/Fakebook.Profile.RestApi/Controllers/ProfilePictureController.cs:68]: asp.net core has a built-in filter for this type of thing. there's a default request size limit of 30 MB or so (though it's been disabled on this method), and if you want to lower that, then look up RequestSizeLimitAttribute.

but  this should have been done with TDD into this branch. so start over with this change - add failing test(s), then make the tests pass in a separate PR into the feature branch.

File: Fakebook.Profile/Fakebook.Profile.UnitTests/DomainTests/ProfileModelTests.cs [hunk 2]
```diff
@@ -283,5 +285,41 @@ public void EarlierTodayIsValid()
...
+        {
+            // arrange
+            DomainProfile profile = new("emailtest@gmail.com", "Bob", "Fields");
+            string followerEmail = "testermail@gmail.com";
+
+            // act
+            // profile.AddFollower(followerEmail);
+
+            // assert
+            // Assert.Contains(followerEmail, profile.FollowerEmails);
```
Comment (escalonn) [inline Fakebook.Profile/Fakebook.Profile.UnitTests/DomainTests/ProfileModelTests.cs:303]: commented-out code?

**Review by escalonn — CHANGES_REQUESTED** [review]
follow tdd process; unrelated changes should be in different branches, and new tests should be added only in `tdd-fail` PRs.

File: Fakebook.Profile/Fakebook.Profile.Domain/Fakebook.Profile.Domain.csproj [hunk 3]
```diff
@@ -4,8 +4,4 @@
     <TargetFramework>net5.0</TargetFramework>
   </PropertyGroup>
 
-  <ItemGroup>
-    <ProjectReference Include="..\Fakebook.Profile.DataAccess\Fakebook.Profile.DataAccess.csproj" />
-  </ItemGroup>
-
```
[the remark this answers is not in the document]
Comment (tdunbar123) [reply Fakebook.Profile/Fakebook.Profile.Domain/Fakebook.Profile.Domain.csproj:10]: We moved the ProfileRepository.cs from the Domain project to the DataAccess project

**Pull request closed without merging**